client/src/Utils/Address/AddressUtils.test.tsx

```diff
@@ -0,0 +1,61 @@
+import axios from 'axios';
+import MockAdapter from 'axios-mock-adapter';
+
+import { getStreetByCity } from 'Utils/Address/AddressUtils';
+import flushPromises from '../Testing/flushPromises';
+
+const mockAxios = new MockAdapter(axios);
+
+describe('getStreetByCity', () => {
+    it('should return Avnei Chefetz streets', async () => {
+        const avniChefetzStreets = [
+            { displayName: "אודם", id: "1038" },
+            { displayName: "איבי הנחל", id: "1820" }
+        ];
+
+        let expectedMapAvniHefetz = new Map();
+        expectedMapAvniHefetz.set('1038', avniChefetzStreets[0]);
+        expectedMapAvniHefetz.set('1820', avniChefetzStreets[1]);
+        const avniChefetzCityId = '3793';
+
+        mockAxios.onGet(`/addressDetails/city/${avniChefetzCityId}/streets`).reply(200, avniChefetzStreets);
+
+        const mockcallback = jest.fn()
+        getStreetByCity(avniChefetzCityId, mockcallback);
+        await flushPromises();
+        expect(mockcallback).toBeCalled()
+        expect(mockcallback).toHaveBeenCalledWith(expectedMapAvniHefetz);
+    });
+
+    it('should return undefined', async () => {
+        const admitStreets = [{}];
+
+        let expectedMapAdmit = new Map();
+        expectedMapAdmit.set(undefined, {});
+        const admitCityId = '1068';
+
+        mockAxios.onGet(`/addressDetails/city/${admitCityId}/streets`).reply(200, admitStreets);
```

Comment:
putting the onGet inline is really clever 🚀 , however, consider adding something like afterEach(mock.reset()) or using replyOnce instead of reply to make sure that the axios mock isn't 'corrupted'